tests/Doctrine/Tests/DBAL/Functional/Schema/SchemaManagerFunctionalTestCase.php

```diff
@@ -1192,4 +1192,19 @@ public function testDoesNotListIndexesImplicitlyCreatedByForeignKeys()
         self::assertArrayHasKey('explicit_fk1_idx', $indexes);
         self::assertArrayHasKey('idx_3d6c147fdc58d6c', $indexes);
     }
+
+    public function testEscapedDefaultValueMustBePreserved()
+    {
+        $value = "a\\0b\\'c\"d\te\\Zf\\\\g''h";
```

Comment:
@PowerKiKi 

If you agree, I propose to test a minimal set of escape sequences in `SchemaManagerFunctionalTestCase`. The current test will fail on SQLServer for example (think about \Z - https://docs.microsoft.com/en-us/sql/integration-services/expressions/numeric-string-and-boolean-literals)

My idea (feel free to implement or not as you think it's best), is to create a method in the base class `SchemaManagerFunctionalTestCase` with those specific escape sequences:

```php
protected function getEscapedLiteralSequence(): string
        return implode('+', [
            "\\'",          // A single quote
            '\\"',          // A double quote
            '\\n',          // A new-line character
            '\\r',          // A carriage return character
            '\\\\',         // A backslash (\) character
        ];
}
```
And change the line above into

```php
   $value = $this->getEscapedLiteralSequence()
```

See my next comment in `AbstractMySQLPlatformTestCase` to test the full MySQL sequences set.